albam/blender_ui/tools.py

```diff
@@ -76,6 +78,32 @@ def draw(self, context):
             text="",)
 
 
+@blender_registry.register_blender_type
+class ALBAM_PT_VGMerger(bpy.types.Panel):
+    '''UI Tool subpanel in Mesh Object Data'''
+    bl_label = "Vertex Groups Merger (Albam)"
+    bl_idname = "ALBAM_PT_VGMerger"
+    bl_space_type = 'PROPERTIES'
+    bl_region_type = 'WINDOW'
+    bl_context = "data"
+    CONTEXT_ITEM_NAME = "mesh"
+
+    def draw(self, context):
+        layout = self.layout
+
+        scn = context.scene.albam.tools_settings
+        row = layout.row()
+        row.prop_search(scn, "vg_a", context.active_object, "vertex_groups", text="Group A")
+        row = layout.row()
+        row.prop_search(scn, "vg_b", context.active_object, "vertex_groups", text="Group B")
+        row = layout.row()
+        row.operator("albam.vg_merge")
+
+    @classmethod
+    def poll(cls, context):
```

Comment:
Adding this panel in the mesh section unconditionally feels a little invasive. It's not a core part of the add-on after all.
If this works on the active object, couldn't it be added to the tools section under Albam's own panel?

Reply:
There is an issue with **prop_search**, when no object is selected it raises an error because of the None type, so to move it on the addon's panel something smart should be done, like don't try to draw them if the selection is empty or so.

Reply:
In the tools panel case, what about using the poll function to return `True `only when `context.active_object` is not `None`? That should avoid the draw call and hence the traceback.

For the other case, keeping it in the mesh data section, one could use the same logic in the poll function that [is used for custom props materials/meshes](https://github.com/Brachi/albam/blob/35c750cb99496a2207fc2f93b963bafcc5553a80/albam/blender_ui/custom_properties.py#L232), and only use it if the mesh is part of an "albam asset". 

I personally still prefer it in the tools section, but you use it more than me, so probably know best where to put it. I just want to avoid polluting the UI when the add-on is not used.